packages/slice-machine/src/legacy/lib/models/common/widgets/Link/components.tsx

```diff
@@ -0,0 +1,67 @@
+import { Box, Icon, Tooltip } from "@prismicio/editor-ui";
+import React, { useEffect, useState } from "react";
+import { Checkbox, Flex, Label } from "theme-ui";
+
+import { Col } from "@/legacy/components/Flex";
+
+type DisplayTextModel = { type: string };
+
+interface DisplayTextCheckboxProps {
+  text?: DisplayTextModel;
+  height?: 130 | 127;
+  setFieldValue: (
+    a: string,
+    b?: DisplayTextModel,
+    // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents, @typescript-eslint/no-explicit-any
+  ) => void | Promise<void | any>;
+}
+
+export function DisplayTextCheckbox(props: DisplayTextCheckboxProps) {
+  const { text, height = 130, setFieldValue } = props;
+
+  const [allowDisplayText, setAllowDisplayText] = useState(Boolean(text));
+
+  useEffect(() => {
+    void setFieldValue(
+      "config.text",
+      allowDisplayText ? { type: "Text" } : undefined,
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [allowDisplayText]);
```

Comment:
Instead of doing this, let's call `setFieldValue` in the checkbox's on change, and derive the boolean for the checkbox. That will remove the effect, the extra state and the extra render.

Reply:
That's better, yeah. Done ✅